packages/devtools_app/lib/src/charts/chart.dart

```diff
@@ -20,35 +35,43 @@ class Chart extends StatefulWidget {
   final ChartController controller;
 
   @override
-  ChartState createState() => ChartState();
+  ChartState createState() => ChartState(controller);
 }
 
 class ChartState extends State<Chart> with AutoDisposeMixin {
-  ChartController controller;
+  ChartState(this.controller);
+
+  final ChartController controller;
+
+  /// Helper to hookup notifiers.
+  void _initSetup() {
+    addAutoDisposeListener(controller.traceChanged, () {
+      setState(() {
+        controller.computeChartArea();
+      });
+    });
+  }
 
   @override
   void initState() {
     super.initState();
-    controller = widget.controller;
-  }
 
-  @override
-  void dispose() {
-    controller.dispose();
-    super.dispose();
+    _initSetup();
   }
 
   @override
-  void didChangeDependencies() {
-    super.didChangeDependencies();
+  void didUpdateWidget(Chart oldWidget) {
+    super.didUpdateWidget(oldWidget);
 
-    cancel();
+    if (oldWidget.controller != controller) {
+      _initSetup();
+    }
+  }
 
-    addAutoDisposeListener(controller.traceChanged, () {
-      setState(() {
-        controller.computeChartArea();
-      });
-    });
+  @override
+  void dispose() {
+    controller.dispose();
```

Comment:
disposing the controller from the State object seems wrong. Controllers should have lifecycles independent of what widget is rendering them. Why do we need this?

Reply:
We don't nor the dispose. Removed.

Also, I have a test app that I use noticed there was a problem with state/controller fixed.